src/librustc/hir/lowering/item.rs

```diff
@@ -282,29 +284,31 @@ impl LoweringContext<'_> {
                 self.lower_use_tree(use_tree, &prefix, id, vis, ident, attrs)
             }
             ItemKind::Static(ref t, m, ref e) => {
+                let ty = self.lower_ty(
+                    t,
+                    if self.sess.features_untracked().impl_trait_in_bindings {
+                        ImplTraitContext::OpaqueTy(None)
+                    } else {
+                        ImplTraitContext::Disallowed(ImplTraitPosition::Binding)
+                    }
+                );
                 hir::ItemKind::Static(
-                    self.lower_ty(
-                        t,
-                        if self.sess.features_untracked().impl_trait_in_bindings {
-                            ImplTraitContext::OpaqueTy(None)
-                        } else {
-                            ImplTraitContext::Disallowed(ImplTraitPosition::Binding)
-                        }
-                    ),
+                    self.arena.alloc(ty.into_inner()),
                     m,
                     self.lower_const_body(e),
                 )
             }
             ItemKind::Const(ref t, ref e) => {
+                let ty = self.lower_ty(
+                    t,
+                    if self.sess.features_untracked().impl_trait_in_bindings {
+                        ImplTraitContext::OpaqueTy(None)
+                    } else {
+                        ImplTraitContext::Disallowed(ImplTraitPosition::Binding)
+                    }
+                );
                 hir::ItemKind::Const(
-                    self.lower_ty(
-                        t,
-                        if self.sess.features_untracked().impl_trait_in_bindings {
-                            ImplTraitContext::OpaqueTy(None)
-                        } else {
-                            ImplTraitContext::Disallowed(ImplTraitPosition::Binding)
-                        }
-                    ),
+                    self.arena.alloc(ty.into_inner()),
```

Comment:
This feels noisy and repetitive. Could we bake `self.arena.alloc(ty.into_inner())` into `lower_ty` or if not, for whatever reason, add a `lower_ty_arena`?

Reply:
I'd expect that to happen on the 3rd PR which moves `Ty` to the arena. I guess @cjgillot moved the fields of each type to the arena too so these show up as temporary changes.

Reply:
There doesn't seem to be that many fields that's moved to `&'hir Ty` though. I'd just keep `P<Ty>` for this PR to avoid unnecessary changes.

Reply:
Indeed, a lot of those disappear in #66942. I plan to do the final cleanup in PR 4/4.

Reply:
After discussing with @pnkfelix, I remembered another reason to migrate eagerly the `P<Ty>` into `&'hir Ty`. If I do not do that, I get unused lifetime parameter errors in intermediate commits (ex: `TraitItemKind`). This keeps me from splitting in compiling commits.